src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -310,8 +315,212 @@ ss::future<> snc_quota_manager::quota_balancer_step() {
 }
 
 ss::future<> snc_quota_manager::quota_balancer_update() {
-    vlog(klog.trace, "qb - Quota balancer update: {}", _shard_quotas_update);
-    return ss::make_ready_future<>();
+    inoutpair<collect_dispense_amount_t> shard_quotas_update;
+    std::swap(_shard_quotas_update, shard_quotas_update);
+    vlog(klog.trace, "qb - Update: {}", shard_quotas_update);
+
+    // deliver shard quota updates of value type
+    if (
+      shard_quotas_update.in.amount_is == collect_dispense_amount_t::value
+      || shard_quotas_update.out.amount_is
+           == collect_dispense_amount_t::value) {
+        co_await container().invoke_on_all([shard_quotas_update](
+                                             snc_quota_manager& qm) {
+            to_each(
+              [](
+                bottomless_token_bucket& shard_quota,
+                const collect_dispense_amount_t& shard_quotas_update) {
+                  if (
+                    shard_quotas_update.amount_is
+                    == collect_dispense_amount_t::value) {
+                      shard_quota.set_quota(shard_quotas_update.amount);
+                  }
+              },
+              qm._shard_quota,
+              shard_quotas_update);
```

Comment:
this is confusing because the return value of to_each is thrown away and it seems you only care about the side effect of the lambda that is applied. in this case it seems like it would be far simpler to have a lambda function and call it on shard_quota.{in,out} etc...?